code/game/dna/dna2.dm

```diff
@@ -102,112 +140,152 @@ GLOBAL_LIST_EMPTY(bad_blocks)
+	SetDNAValueRange(DNA_UI_BODY_MARK_B,	color2B(character.m_colours["body"]),	255,	DNA_UI,	TRUE)
 
-	SetUIValueRange(DNA_UI_TAIL_MARK_R,	color2R(character.m_colours["tail"]),	255,	1)
-	SetUIValueRange(DNA_UI_TAIL_MARK_G,	color2G(character.m_colours["tail"]),	255,	1)
-	SetUIValueRange(DNA_UI_TAIL_MARK_B,	color2B(character.m_colours["tail"]),	255,	1)
+	SetDNAValueRange(DNA_UI_TAIL_MARK_R,	color2R(character.m_colours["tail"]),	255,	DNA_UI,	TRUE)
+	SetDNAValueRange(DNA_UI_TAIL_MARK_G,	color2G(character.m_colours["tail"]),	255,	DNA_UI,	TRUE)
+	SetDNAValueRange(DNA_UI_TAIL_MARK_B,	color2B(character.m_colours["tail"]),	255,	DNA_UI,	TRUE)
 
-	SetUIValueRange(DNA_UI_SKIN_TONE,	35-character.s_tone,	220,	1) // Value can be negative.
+	SetDNAValueRange(DNA_UI_SKIN_TONE,	35-character.s_tone,	220,	DNA_UI,	TRUE) // Value can be negative.
 
-	/*SetUIValueRange(DNA_UI_BACC_STYLE,	bodyacc,	GLOB.facial_hair_styles_list.len,	1)*/
-	SetUIValueRange(DNA_UI_HEAD_MARK_STYLE,	head_marks,		GLOB.marking_styles_list.len,		1)
-	SetUIValueRange(DNA_UI_BODY_MARK_STYLE,	body_marks,		GLOB.marking_styles_list.len,		1)
-	SetUIValueRange(DNA_UI_TAIL_MARK_STYLE,	tail_marks,		GLOB.marking_styles_list.len,		1)
+	/*SetDNAValueRange(DNA_UI_BACC_STYLE,	bodyacc,	GLOB.facial_hair_styles_list.len,	DNA_UI,	TRUE)*/
+	SetDNAValueRange(DNA_UI_HEAD_MARK_STYLE,	head_marks,		GLOB.marking_styles_list.len,		DNA_UI,	TRUE)
+	SetDNAValueRange(DNA_UI_BODY_MARK_STYLE,	body_marks,		GLOB.marking_styles_list.len,		DNA_UI,	TRUE)
+	SetDNAValueRange(DNA_UI_TAIL_MARK_STYLE,	tail_marks,		GLOB.marking_styles_list.len,		DNA_UI,	TRUE)
 
 	//Set the Gender
 	switch(character.gender)
 		if(FEMALE)
-			SetUITriState(DNA_UI_GENDER, DNA_GENDER_FEMALE, 1)
+			SetDNATriState(DNA_UI_GENDER, DNA_GENDER_FEMALE, DNA_UI, TRUE)
 		if(MALE)
-			SetUITriState(DNA_UI_GENDER, DNA_GENDER_MALE, 1)
+			SetDNATriState(DNA_UI_GENDER, DNA_GENDER_MALE, DNA_UI, TRUE)
 		if(PLURAL)
-			SetUITriState(DNA_UI_GENDER, DNA_GENDER_PLURAL, 1)
+			SetDNATriState(DNA_UI_GENDER, DNA_GENDER_PLURAL, DNA_UI, TRUE)
 
 
-	UpdateUI()
+	UpdateDNA(DNA_UI)
 
-// Set a DNA UI block's raw value.
-/datum/dna/proc/SetUIValue(block, value, defer = FALSE)
-	if(block <= 0)
+
+/datum/dna/proc/ValidCheck(block, dna_type)
+	if(block <= 0 || dna_type < 0 || dna_type > 2 || dna_type == null)
+		return FALSE
+	else
+		return TRUE
+
+///////////////////////////////////////
+// SET and GET values for DNA blocks
+///////////////////////////////////////
+
+// Set a DNA block's raw value.
+/datum/dna/proc/SetDNAValue(block, value, dna_type, defer = FALSE)
+	if(!ValidCheck(block, dna_type))
 		return
 	ASSERT(value > 0)
 	ASSERT(value <= 4095)
-	UI[block]=value
-	dirtyUI = 1
+	switch(dna_type)
+		if(DNA_UI)
+			UI[block] = value
+			dirtyUI = TRUE
+		if(DNA_SE)
+			SE[block] = value
+			dirtySE = TRUE
+		if(DNA_RP)
+			RP[block] = value
+			dirtyRP = TRUE
 	if(!defer)
-		UpdateUI()
+		UpdateDNA(dna_type)
 
```

Comment:
@blinkdog Calling myself out from earlier. You were correct on passing `defer`. Yikes.